### PR TITLE
Comments round

### PR DESCRIPTION
**Pull request by Korsarius**

### CODE REVIEW

File: src/app/app.component.html [hunk 1]
```diff
@@ -1,6 +1,14 @@
 <h1 class="mat-display-1">Вікторина</h1>
 <div *ngIf="teamInAction() as team">
-  <p>Грає команда - <b>{{team.name}}</b></p>
+  <p>Грає команда - <b>{{ team.name }}</b></p>
 </div>
 <app-enter-team-names *ngIf="round() === null"></app-enter-team-names>
 <app-trick *ngIf="round() === roundName.Trick"></app-trick>
+<app-comments *ngIf="round() === roundName.Comments"></app-comments>
+
+<footer *ngIf="round() !== null" class="footer">
```
Comment (DimaKomoff) [inline src/app/app.component.html:41]: delete this shit...

File: src/app/store/global/global.state.ts [hunk 2]
```diff
@@ -57,18 +56,17 @@ export class GlobalState {
   @Action(GlobalActions.SetTeamScore)
   setTeamScore(ctx: StateContext<IGlobalState>, action: GlobalActions.SetTeamScore) {
```
Comment (DimaKomoff) [inline src/app/store/global/global.state.ts]: need to change name to updateTeamScore

File: src/app/app.component.html (hunk 1 above)
Comment (Korsarius) [reply src/app/app.component.html:41]: Can I leave it for now, for debugging only, please?

File: src/app/interfaces/comments-store.intarface.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  currentQuestion: Question;
+}
+
+export type ICommentsRoundState = {
+  [key in Team]: IQuestionsExtended;
+};
+
+export interface ICommentQuestion {
+  comments: string[];
+  score: number;
```
Comment (DimaKomoff) [inline src/app/interfaces/comments-store.intarface.ts:18]: need to create a mapper for this

File: src/app/interfaces/comments-store.intarface.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  step: number;
```
Comment (DimaKomoff) [inline src/app/interfaces/comments-store.intarface.ts:19]: should be Enum

File: src/app/interfaces/comments-store.intarface.ts [hunk 5]
```diff
@@ -0,0 +1,26 @@
+import { Team } from '../enums/global-state.enum';
+import { Question } from '../enums/comments.enum';
+
+type Questions = {
+  [key in Question]: ICommentQuestion
+};
+
+interface IQuestionsExtended extends Questions {
```
Comment (DimaKomoff) [inline src/app/interfaces/comments-store.intarface.ts:8]: redundant interface

File: src/app/constants/comments-store.constant.ts [hunk 6]
```diff
@@ -0,0 +1,154 @@
+import { Team } from '../enums/global-state.enum';
+import { ICommentsRoundState } from '../interfaces/comments-store.intarface';
+import { Question } from '../enums/comments.enum';
+
+export const COMMENTS_ROUND_STORE_CONSTANT = {
+  localStorageKey: 'COMMENTS_ROUND_STORE',
+}
+
+export const INITIAL_DEFAULT_COMMENTS_STATE: ICommentsRoundState = {
```
Comment (DimaKomoff) [inline src/app/constants/comments-store.constant.ts:9]: to many extra data

File: src/app/components/comments/comments.component.ts [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+  constructor(private _snackBar: MatSnackBar) {}
+
+  ngOnInit(): void {
+    const localStorageState = this._localStorage.getItem(COMMENTS_ROUND_STORE_CONSTANT.localStorageKey);
+    if (localStorageState) {
+      this._store.dispatch(new CommentsRoundActions.SetInitialState(JSON.parse(localStorageState)));
+    }
+  }
+
+  checkAnswer(team: Team, option: ICommentOption): void {
```
Comment (DimaKomoff) [inline src/app/components/comments/comments.component.ts:48]: you have all business logic inside the component. If we use redux we need to write logic in effects(in ngxs it's @Action() methods)

File: src/app/store/comments/comments.state.ts [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+  }
+
+  @Action(CommentsRoundActions.ChangeTeamQuestion)
+  changeTeamQuestion(ctx: StateContext<ICommentsRoundState>, action: CommentsRoundActions.ChangeTeamQuestion): void {
+    const state = ctx.getState();
+    const { team, currentQuestion } = action;
+    const questions: Question[] = (Object.values(Question).filter(q => typeof q === 'number')) as Question[];
+    const currentQuestionIndex: number = questions.findIndex(q => q === currentQuestion);
+    const nextQuestion: Question = questions[currentQuestionIndex + 1];
+    console.log('nextQuestion: ', nextQuestion);
```
Comment (DimaKomoff) [inline src/app/store/comments/comments.state.ts:79]: remove

**Pull request merged**